packages/mobile/src/components/challenge-rewards-drawer/ChallengeRewardsDrawerProvider.tsx

```diff
@@ -10,35 +10,19 @@ import {
   getClaimStatus
 } from 'audius-client/src/common/store/pages/audio-rewards/selectors'
 import {
-  ChallengeRewardsModalType,
   claimChallengeReward,
   ClaimStatus,
   resetAndCancelClaimReward
 } from 'audius-client/src/common/store/pages/audio-rewards/slice'
 import { setVisibility } from 'audius-client/src/common/store/ui/modals/slice'
 import { Maybe } from 'audius-client/src/common/utils/typeUtils'
-import {
-  ACCOUNT_VERIFICATION_SETTINGS_PAGE,
-  AUDIO_PAGE,
-  TRENDING_PAGE,
-  UPLOAD_PAGE
-} from 'audius-client/src/utils/route'
-import { isEqual } from 'lodash'
-import { ImageSourcePropType } from 'react-native'
+import { AUDIO_PAGE } from 'audius-client/src/utils/route'
 
-import Headphone from 'app/assets/images/emojis/headphone.png'
-import IncomingEnvelope from 'app/assets/images/emojis/incoming-envelope.png'
-import LoveLetter from 'app/assets/images/emojis/love-letter.png'
-import MobilePhoneWithArrow from 'app/assets/images/emojis/mobile-phone-with-arrow.png'
-import MultipleMusicalNotes from 'app/assets/images/emojis/multiple-musical-notes.png'
-import WhiteHeavyCheckMark from 'app/assets/images/emojis/white-heavy-check-mark.png'
-import IconArrow from 'app/assets/images/iconArrow.svg'
-import IconCheck from 'app/assets/images/iconCheck.svg'
-import IconUpload from 'app/assets/images/iconUpload.svg'
 import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
 import { usePushRouteWeb } from 'app/hooks/usePushRouteWeb'
 import { useRemoteVar } from 'app/hooks/useRemoteConfig'
 import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+import { challengesConfig } from 'app/utils/challenges'
 
 import Button, { ButtonType } from '../button'
```

Comment:
would it be too much of a lift to us the core button?

Reply:
I can do this as a sep PR, this isn't new here!